accepted/0000-registry-url-in-lock-file.md

```diff
@@ -0,0 +1,71 @@
+- Start Date: 2017-09-21
+- RFC PR:
+- Yarn Issue:
+
+# Summary
+
+The yarn.lock file includes information about the registry that it was generated against. It would
+be nice if there was no registry information associated with the lock file
+
+# Motivation
+
+We currently have two internal registries/mirrors. We generate the lock file against one we have in
+our development environment. When our code is deployed to our build/testing environment, we
+no longer have access to the original registry that was used to generate the lock file. This causes
+our download to error as yarn cannot retrieve the packages.
+
+The expected output is for yarn to recognise the registry URL supplied (via either cli flag or
+`.yarnrc` or `.npmrc`) and download from that if the registry used in the lock file is not available.
+
+
+# Detailed design
+
+Let me know if this isn't enough, first time writing an RFC.
+
+As a dev that generated their lock file against REGISTRY_A
+I would like to pass a cli flag designating REGISTRY_B
+And I would like yarn to download my dependencies from REGISTRY_B
+
+As a dev that generated their lock file against REGISTRY_A
+I would like to designate an alternate registry URL in my `.yarnrc` or `.npmrc` (REGISTRY_B)
+And I would like yarn to download my dependencies from the alternate URL (REGISTRY_B)
+
+I feel like the simplest solution would be to add a "hash" or "commit-ish" field to entries.

```

Comment:
That is a lot of text that does not add much value compared to the previous section.
Here we need to see concrete examples how Yarn will behave with the new feature.
I suggest something like this:


If yarn.lock file contains
```
abbrev@1:
   version "1.1.0"
   resolved "https://registry.yarnpkg.com/abbrev/-/abbrev-1.1.0.tgz#d0554c2256636e2f56e7c2e5ad183f859428d81f"
 ```
And .yarnrc has setting `override-registry={"registry.yarn.pkg.com": "registry.npm.org"}`
Then Yarn will fetch `abbrev-1.1.0.tgz` from `https://registry.npm.org/abbrev/-/abbrev-1.1.0.tgz#d0554c2256636e2f56e7c2e5ad183f859428d81f` instead of the one specified in yarn.lock.